javasource/databaseconnector/impl/JdbcConnector.java

```diff
@@ -12,103 +24,114 @@
-	}
-
-	private Function<Object, Object> toSuitableValue(final PrimitiveType type) {
-		return v -> type == PrimitiveType.Binary ? new ByteArrayInputStream((byte[]) v) : v;
-	}
-
-	private Stream<Map<String, Optional<Object>>> executeQuery(final String jdbcUrl, final String userName,
-			final String password, final IMetaObject metaObject, final String sql) throws SQLException {
-		logNode.trace(String.format("executeQuery: %s, %s, %s", jdbcUrl, userName, sql));
-
-		try (Connection connection = connectionManager.getConnection(jdbcUrl, userName, password);
-				PreparedStatement preparedStatement = connection.prepareStatement(sql);
-				ResultSet resultSet = preparedStatement.executeQuery()) {
-			ResultSetReader resultSetReader = new ResultSetReader(resultSet, metaObject);
-
-			return resultSetReader.readAll().stream();
-		}
-	}
-
-	public long executeStatement(final String jdbcUrl, final String userName, final String password, final String sql)
-			throws SQLException {
-		logNode.trace(String.format("executeStatement: %s, %s, %s", jdbcUrl, userName, sql));
-
-		try (Connection connection = connectionManager.getConnection(jdbcUrl, userName, password);
-				PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
-			return preparedStatement.executeUpdate();
-		}
-	}
+    private final ILogNode logNode;
+    private final ObjectInstantiator objectInstantiator;
+    private final ConnectionManager connectionManager;
+    private final PreparedStatementCreator preparedStatementCreator;
+
+    public JdbcConnector(final ILogNode logNode, final ObjectInstantiator objectInstantiator,
+                         final ConnectionManager connectionManager, final PreparedStatementCreator preparedStatementCreator) {
+        this.logNode = logNode;
+        this.objectInstantiator = objectInstantiator;
+        this.connectionManager = connectionManager;
+        this.preparedStatementCreator = preparedStatementCreator;
+    }
+
+    public JdbcConnector(final ILogNode logNode) {
+        this(logNode,
+                new ObjectInstantiatorImpl(),
+                ConnectionManagerSingleton.getInstance(),
+                new PreparedStatementCreatorImpl());
+    }
+
+    public Stream<IMendixObject> executeQuery(final String jdbcUrl, final String userName, final String password,
+                                              final IMetaObject metaObject, final String sql, final IContext context) throws SQLException {
+        logNode.trace(String.format("executeQuery: %s, %s, %s", jdbcUrl, userName, sql));
+
+        try (Connection connection = connectionManager.getConnection(jdbcUrl, userName, password);
+             PreparedStatement preparedStatement = preparedStatementCreator.create(sql, connection);
+             ResultSet resultSet = preparedStatement.executeQuery()) {
+            ResultSetReader resultSetReader = new ResultSetReader(resultSet, metaObject);
```

Comment:
Something odd is going on with formatting over here.

Reply:
The formatting rules across files are not consistent. I'll reformat all files using the runtime project rules in a separate PR. I'll revert to old formatting for the sake of readability here.